control-plane/subcommand/server-acl-init/command_test.go

```diff
@@ -2192,6 +2153,18 @@ func TestRun_PoliciesAndBindingRulesForACLLogin_PrimaryDatacenter(t *testing.T)
 			PolicyNames: []string{"client-policy"},
 			Roles:       []string{resourcePrefix + "-client-acl-role"},
 		},
+		{
+			TestName: "Terminating Gateway",
+			TokenFlags: []string{"-terminating-gateway-name=terminating",
+				"-terminating-gateway-name=gateway",
+				"-terminating-gateway-name=another-gateway"},
+			PolicyNames: []string{"terminating-terminating-gateway-policy",
+				"gateway-terminating-gateway-policy",
+				"another-gateway-terminating-gateway-policy"},
+			Roles: []string{"release-name-consul-terminating-acl-role",
```

Comment:
Are there any scenarios besides additional data centers where someone can define a duplicate gateway name? For example, could I define:
- terminating gateway named foo (where role would be `release-name-consul-foo-acl-role`)
- ingress gateway named foo (where role would be `release-name-consul-foo-acl-role`)

This seems like a possible issue?

Reply:
I think it makes sense to update the role name logic to be similar to how the policy names are generated. This could be a problem AFAIR.

minor nit: can we swap out `release-name-consul` for `resourcePrefix` like in the other fixtures? 🙏